dashboard/src/utils/filterDataset.js

```diff
@@ -1,9 +1,7 @@
 import { constructUTCDate } from "./constructDate";
 export const filterData = (dataArray, startDate, endDate, controllerValue) => {
   let res = dataArray.filter((data) => {
-    let datasetDate = constructUTCDate(
-      new Date(data.metadata["dataset.created"].split(":")[0])
-    );
+    let datasetDate = constructUTCDate(`${data.metadata["dataset.created"]}Z`);
```

Comment:
This is great, but obviously we need to coordinate with Dave on when the server will start producing full-ISO-formatted values, at which point this code will need to be modified again.

Reply:
Yes, well, the code is in; so at this point it's a matter of when I upgrade my server VM. We had deliberately negotiated deferring that upgrade until after this PR went in. We can obviously change that decision at any time.

Reply:
Ideally, the contents of the repo should be self-consistent; thus, now that the server change is "in", this code should match it when it is merged...but, I've lost my grip on how much of the ideal we're supposed to be striving for with this PR.

In point of fact, Shubham has to be able to test his code.  If it's a straightforward process, I would suggest that Dave update his server and at the same time Shubham removes the `Z` above, and then it will all just work, right?

Reply:
@shubham-html-css-js is going to revisit the implementation of `constructUTCDate()`, so, @dbutenhof, you might as well proceed with updating your test server, and he can address this with the other changes.

Reply:
> @shubham-html-css-js is going to revisit the implementation of `constructUTCDate()`, so, @dbutenhof, you might as well proceed with updating your test server, and he can address this with the other changes.

I think that's probably a good point. For better or worse, I decided to update my server:

```
  {
    "controller": "dhcp31-107.perf.lab.eng.bos.redhat.com",
    "metadata": {
      "dashboard.saved": null,
      "dataset.access": "public",
      "dataset.created": "2022-02-16T13:21:29+00:00",
      "dataset.owner": "david.butenhof@redhat.com",
      "server.deletion": "2024-02-17"
    },
```

(Note that `server.deletion` ought to be ISO 8601 as well, however that's a *string* value defined when the dataset is uploaded and I'm not doing anything to "fix" that on update... so we'd have to upload a new dataset to see it happen. You can however see the change in `dataset.created`.)